std-bits/table/src/main/java/org/enso/table/data/column/storage/type/IntegerType.java

```diff
@@ -69,6 +70,10 @@ public boolean fits(BigInteger value) {
     }
   }
 
+  public boolean fits(BigDecimal value) {
+    return fits(value.toBigInteger());
+  }
```

Comment:
This is OK, but I'm not a fan of this.

If you look at the implementation of `BigDecimal::longValue` you will see that (except for a fast path for values close to 0), it calls `toBigInteger` and then calls `BigInteger::longValue` on the result.

That means that when doing the `BigDecimal` -> `Integer` type cast, we will be calling `toBigInteger` _twice_ for every value that fits - once in the `fits` method and second time in the `longValue` method when appending. If `toBigInteger` was a very cheap operation that would be OK - but it actually seems relatively expensive. So I'm wondering if it would make sense to change the logic in `ToIntegerStorageConverter` and first do a `toBigInteger` on the value and only later call `fits` and `longValue` on that - ensuring we convert `toBigInteger` always only once.

Reply:
Done

Reply:
Done